### PR TITLE
Return early in case of a faulty sender

### LINKED ISSUE #4206: Return early in Zug's handle_sync_response if the sender is faulty.
When a sender is faulty, we should stop every message treatment and return a "disconnect from this node" instruction. This allows us not to spend too much time on the messages from a not-previously-known malicious actor.

### PR DESCRIPTION
**Pull request by real-felix**
Fix #4206

### CODE REVIEW

File: node/src/components/consensus/protocols/zug.rs [hunk 1]
```diff
@@ -2416,6 +2444,14 @@ where
     }
 }
 
+fn outcomes_or_disconnect<C: Context>(
+    result: Result<ProtocolOutcomes<C>, FaultySender>,
+) -> ProtocolOutcomes<C> {
+    result.unwrap_or_else(|sender| vec![ProtocolOutcome::Disconnect(sender.0)])
+}
+
+struct FaultySender(NodeId);
```
Comment (fizyk20) [inline node/src/components/consensus/protocols/zug.rs:2453]: Nit: I'd move it somewhere nearer the top of the file.

**Review by fizyk20 — APPROVED** [review]
Looks good :+1:

File: node/src/components/consensus/protocols/zug.rs (hunk 1 above)
Comment (real-felix) [reply node/src/components/consensus/protocols/zug.rs:2453]: I'll do that in another PR, then, I won't create a commit only for moving a line.

**Pull request merged**